substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -1086,34 +1051,38 @@ impl<'a, E: Ext, M: ?Sized + Memory<E::T>> Runtime<'a, E, M> {
 			let salt: [u8; 32] = memory.read_array(salt_ptr)?;
 			Some(salt)
 		};
-		let instantiate_outcome = self.ext.instantiate(
+
+		match self.ext.instantiate(
 			weight,
 			deposit_limit,
 			code_hash,
 			value,
 			input_data,
 			salt.as_ref(),
-		);
-		if let Ok((address, output)) = &instantiate_outcome {
-			if !output.flags.contains(ReturnFlags::REVERT) {
-				self.write_fixed_sandbox_output(
+		) {
+			Ok(address) => {
+				if !self.ext.last_frame_output().flags.contains(ReturnFlags::REVERT) {
+					write_fixed_sandbox_output::<E, M>(
+						memory,
+						address_ptr,
+						&address.as_bytes(),
+						true,
+					)?;
+				}
+				self.charge_gas(RuntimeCosts::CopyToContract(
```

Comment:
if you pass SENTINEL here then you still charge gas here don't you?

Reply:
Ah yes good catch! Need to account for that everywhere :-/

Reply:
I just decided to revert the changes related to the sandbox write helpers. It was a drive-by anyways.